snap/quota/quota.go

```diff
@@ -271,11 +272,31 @@ func (grp *Group) getAllowedCPUs() []int {
 	return grp.CPULimit.AllowedCPUs
 }
 
+// GetCorrectedCPUCount returns the maximum number of allowed CPU cores for
+// this group. It needs to take into account that CPU set might have been set
+// to limit the number of cores, or a direct limit on the number of cores.
+// Goal is to select the most restrictive limit.
+func (grp *Group) GetCorrectedCPUCount() int {
+	if grp.CPULimit == nil {
+		return 0
```

Comment:
This will be taken into account in the next PR https://github.com/snapcore/snapd/pull/11607

Reply:
Similar to cgroup check previously, should we check runtime.NumCPU in the client? Sounds like snapd-side check again?

Reply:
I don't believe it to be neccessary, this function is only used to get the most restrictive cpu count for a group, up to a maximum of the number of cores. This is used when calculating the space left in a quota group that has a cpu quota

Reply:
Ah sorry, ignore my remark, I thought I was looking at cmd/snap code.

Reply:
Haha no issues! I think the explaination is good to have anyway!

Reply:
Mmm... now #11607 has been closed; does this PR too need investigation? If not, then please move the part that fixes the `return 0` into this branch, so that we can merge it.